sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/QueueClientBuilder.java

```diff
@@ -0,0 +1,338 @@
+import com.azure.core.amqp.implementation.AzureTokenManagerProvider;
+import com.azure.core.amqp.implementation.CbsAuthorizationType;
+import com.azure.core.amqp.implementation.ConnectionOptions;
+import com.azure.core.amqp.implementation.ConnectionStringProperties;
+import com.azure.core.amqp.implementation.MessageSerializer;
+import com.azure.core.amqp.implementation.ReactorHandlerProvider;
+import com.azure.core.amqp.implementation.ReactorProvider;
+import com.azure.core.amqp.implementation.StringUtil;
+import com.azure.core.amqp.implementation.TokenManagerProvider;
+import com.azure.core.amqp.implementation.TracerProvider;
+import com.azure.core.annotation.ServiceClientBuilder;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.exception.AzureException;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.tracing.Tracer;
+import com.azure.messaging.servicebus.implementation.ServiceBusClientConstants;
+import com.azure.messaging.servicebus.implementation.ServiceBusConnectionProcessor;
+import com.azure.messaging.servicebus.implementation.ServiceBusAmqpConnection;
+import com.azure.messaging.servicebus.implementation.ServiceBusReactorAmqpConnection;
+import com.azure.messaging.servicebus.implementation.ServiceBusSharedKeyCredential;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.util.Map;
+import java.util.Objects;
+import java.util.ServiceLoader;
+
+/***
+ * The builder to create {@link QueueReceiverAsyncClient} and {@link QueueSenderAsyncClient}.
+ */
+@ServiceClientBuilder(serviceClients = { QueueReceiverAsyncClient.class, QueueSenderAsyncClient.class})
+public final class QueueClientBuilder {
+
+    private static final String AZURE_SERVICE_BUS_CONNECTION_STRING = "AZURE_SERVICE_BUS_CONNECTION_STRING";
+    private static final AmqpRetryOptions DEFAULT_RETRY =
+        new AmqpRetryOptions().setTryTimeout(ServiceBusClientConstants.OPERATION_TIMEOUT);
+
+    private static final String SERVICEBUS_PROPERTIES_FILE = "azure-messaging-servicebus.properties";
+    private static final String NAME_KEY = "name";
+    private static final String VERSION_KEY = "version";
+    private static final String UNKNOWN = "UNKNOWN";
+
+    private final ClientLogger logger = new ClientLogger(QueueClientBuilder.class);
+
+    private ProxyOptions proxyOptions;
+    private TokenCredential credentials;
+    private Configuration configuration;
+    private AmqpRetryOptions retryOptions;
+    private Scheduler scheduler;
+    private AmqpTransportType transport = AmqpTransportType.AMQP;
+    private String fullyQualifiedNamespace;
+    private String queueName;
+    private ServiceBusConnectionProcessor servicerBusConnectionProcessor;
```

Comment:
```suggestion
    private ServiceBusConnectionProcessor serviceBusConnectionProcessor;
```
Or go full bore and add more 'r's.